### PR TITLE
Centralize user name and email presentation

### PR DESCRIPTION
**Pull request by mathjazz**
We present users differently in different places. I thought it would be nice to have a centralized place to generate all these display names.

@jotes r?

### CODE REVIEW

File: pontoon/base/models.py [hunk 1]
```diff
@@ -100,8 +100,18 @@ def user_translated_locales(self):
...
+
+
+@property
 def user_display_name(self):
-    name = self.first_name or self.email.split('@')[0]
+    return self.first_name or self.email.split('@')[0]
+
+
+@property
+def user_display_name_and_email(self):
```
Comment (jotes) [inline pontoon/base/models.py]: i would change name to the vcs_display_name to give more context.

File: pontoon/base/models.py [hunk 2]
```diff
@@ -100,8 +100,18 @@ def user_translated_locales(self):
 
 
 @property
+def user_name_or_email(self):
```
Comment (jotes) [inline pontoon/base/models.py]: Somehow, i feel we should use more generic for the methods, because they'll be easier to change in future.

**Comment by jotes** [comment]
r+ but we still need add some kind of profile management for the translators.

File: pontoon/base/models.py (hunk 1 above)
Comment (mathjazz) [inline pontoon/base/models.py]: We also use the name on other occasions, e.g. in PO file headers.

**Comment by mathjazz** [comment]
> r+ but we still need add some kind of profile management for the translators.

You mean custom user model?

File: pontoon/base/models.py (hunk 2 above)
Comment (mathjazz) [inline pontoon/base/models.py]: You mean more generic method names?

**Comment by jotes** [comment]
@mathjazz i'm not sure if user can change first_name on the our side. However, it's probably connected with our transition from the persona to self-hosted accounts.

**Comment by mathjazz** [comment]
They can change their first_name on the /profile page.

**Pull request merged**